internal/versions/versions.go

```diff
@@ -103,6 +106,29 @@ func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params o
 			supportedArchs = []string{*releaseImage.CPUArchitecture}
 		}
 
+		// (MGMT-11859) We are filtering out multiarch release images so that they are available
+		//              only for customers allowed to use them. This is in order to be able to
+		//              expose them in OCP pre-4.13 without making them generally available.
+		if len(supportedArchs) > 1 {
+			hasMultiarchAuthorization := false
+			checkedForMultiarchAuthorization := false
+
+			if !checkedForMultiarchAuthorization {
+				checkedForMultiarchAuthorization = true
+				if err := auth.ValidateAccessToMultiarch(ctx, h.authzHandler); err != nil {
```

Comment:
Maybe we can incorporate the caching in this func.

Reply:
We decided not to do caching for the ARM capability at the time since it was a flow that is not triggered too often. Isn't it the case here as well?

Reply:
i think that it's the same case, if the api is called only on update and create then cache is not needed.

Reply:
iirc, this API is called by the UI on every draft cluster creation, so caching is not crucial but would still be nice to have I think. Anyway, it'll make the func clearer if this logic can be extracted.

Reply:
That's true, the API is called every time you open UI for cluster creation. I just want to highlight the fact that this code is only temporary because in 4.13 everything will be multiarch. So everything I do in this PR will need to be reverted. And probably everyone here agrees that it will be not as simple as a commit-revert. Maybe we should classify it as "not ideal but good enough to keep us running for one release"?

Reply:
Oh ok, if that's the case then it's indeed not crucial:)